shared/src/ledger/ibc/vp/channel.rs

```diff
@@ -823,28 +810,38 @@ where
         key: &(PortId, ChannelId, Sequence),
     ) -> Ics04Result<String> {
         let commitment_key = commitment_key(&key.0, &key.1, key.2);
-        self.get_packet_info(&commitment_key)
-            .map_err(|_| Ics04Error::packet_commitment_not_found(key.2))
+        match self.ctx.read_post(&commitment_key) {
+            Ok(Some(value)) => std::str::from_utf8(&value)
+                .map_err(|_| Ics04Error::implementation_specific())
+                .map(|s| s.to_string()),
+            Ok(None) => Err(Ics04Error::packet_commitment_not_found(key.2)),
+            Err(_) => Err(Ics04Error::implementation_specific()),
+        }
     }
 
     fn get_packet_receipt(
         &self,
         key: &(PortId, ChannelId, Sequence),
     ) -> Ics04Result<Receipt> {
         let receipt_key = receipt_key(&key.0, &key.1, key.2);
+        let expect = PacketReceipt::default().as_bytes().to_vec();
         match self.ctx.read_post(&receipt_key) {
-            Ok(Some(_)) => Ok(Receipt::Ok),
+            Ok(Some(v)) if v == expect => Ok(Receipt::Ok),
             _ => Err(Ics04Error::packet_receipt_not_found(key.2)),
         }
     }
 
+    // TODO should return Vec<u8> or Acknowledgment. fix in ibc-rs?
```

Comment:
should we open an issue for this?

Reply:
Thank you for your suggestion!
I'll recheck the implementation of ibc-go and open an issue.